src/bin/bors.rs

```diff
@@ -1,34 +1,65 @@
 use std::net::SocketAddr;
+use std::path::PathBuf;
 use std::sync::Arc;
 
 use anyhow::Context;
+use axum::extract::State;
 use axum::http::StatusCode;
 use axum::response::IntoResponse;
 use axum::routing::post;
 use axum::Router;
 use clap::Parser;
 use tower::limit::ConcurrencyLimitLayer;
 
-use bors::config::Config;
+use bors::github::api::RepositoryAccess;
+use bors::github::service::{github_webhook_process, WebhookSender};
 use bors::github::webhook::GitHubWebhook;
+use bors::github::WebhookSecret;
 
 #[derive(clap::Parser)]
 struct Opts {
     /// Secret used to authenticate webhooks.
     #[arg(long, env = "WEBHOOK_SECRET")]
     webhook_secret: String,
+    /// Github App ID.
+    #[arg(long, env = "APP_ID")]
+    app_id: u64,
+    /// Path to a private key used to authenticate as a Github App.
+    #[arg(long, env = "PRIVATE_KEY_PATH")]
+    private_key_path: PathBuf,
 }
 
-async fn github_webhook(GitHubWebhook(event): GitHubWebhook) -> impl IntoResponse {
-    println!("Received GitHub webhook: {:?}", event);
-    (StatusCode::OK, "")
+struct ServerState {
+    sender: WebhookSender,
+    webhook_secret: WebhookSecret,
 }
 
-async fn server(config: Config) -> anyhow::Result<()> {
-    let state = Arc::new(config);
+impl AsRef<WebhookSecret> for ServerState {
+    fn as_ref(&self) -> &WebhookSecret {
+        &self.webhook_secret
+    }
+}
```

Comment:
The `AsRef` is here so that the webhook axum extractor doesn't need to know specifically how does the state look like. I can implement the extractor with the knowledge of the server state if you want.

Reply:
I'd implement the extractor with the knowledge of the server state, yeah.